### PR TITLE
Add a CDI producer for json object converters

### PR DESCRIPTION
**Pull request by allanmckenzie**

### CODE REVIEW

File: framework-utilities/test-utils/test-utils-framework-api/pom.xml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-core</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>uk.gov.justice.utils</groupId>
+            <artifactId>test-utils-core</artifactId>
```
Comment (Morbu) [inline framework-utilities/test-utils/test-utils-framework-api/pom.xml]: is this one test scope too? If not maybe group it with the other non test scope dependencies ?

Comment (allanmckenzie) [reply framework-utilities/test-utils/test-utils-framework-api/pom.xml]: Unfortunately we can't make it test scope. It's used by other test libraries and they need to import it as a jar. If it's test scope here those libraries cant' see it

Comment (allanmckenzie) [reply framework-utilities/test-utils/test-utils-framework-api/pom.xml]: Which is a long way of saying that maven is crap and if I could make it test scope I would

File: framework-utilities/utilities/utilities-core/src/main/java/uk/gov/justice/services/common/converter/ObjectToJsonObjectConverter.java [hunk 2]
```diff
@@ -1,34 +1,57 @@
...
+        this.mapper = mapper;
+    }
 
     @Override
     public JsonObject convert(final Object source) {
         try {
-            final JsonObject jsonObject = mapper.readValue(mapper.writeValueAsString(source), JsonObject.class);
+            final String json = mapper.writeValueAsString(source);
+
+            if (json != null) {
```
Comment (Morbu) [inline framework-utilities/utilities/utilities-core/src/main/java/uk/gov/justice/services/common/converter/ObjectToJsonObjectConverter.java:43]: Is this new check going to alter the exception that is throw? I guess it shouldn't actually matter, but just want to confirm.

Comment (allanmckenzie) [reply framework-utilities/utilities/utilities-core/src/main/java/uk/gov/justice/services/common/converter/ObjectToJsonObjectConverter.java:43]: No - it's actually a fix of code that didn't actually work, but the test didn't fail correctly

**Review by Morbu — CHANGES_REQUESTED** [review]

File: framework-utilities/utilities/utilities-core/src/main/java/uk/gov/justice/services/common/converter/ObjectToJsonObjectConverter.java [hunk 3]
```diff
@@ -1,34 +1,57 @@
...
-import javax.inject.Inject;
+import javax.enterprise.inject.Vetoed;
 import javax.json.JsonObject;
 
 import com.fasterxml.jackson.databind.ObjectMapper;
 
 /**
  * Converts a Pojo to a JsonObject
  */
+@Vetoed
```
[the remark this answers is not in the document]
Comment (allanmckenzie) [reply framework-utilities/utilities/utilities-core/src/main/java/uk/gov/justice/services/common/converter/ObjectToJsonObjectConverter.java:17]: yes. Cakeshop blows up if you don't

File: framework-utilities/test-utils/test-utils-framework-api/pom.xml (hunk 1 above)
Comment (Morbu) [reply framework-utilities/test-utils/test-utils-framework-api/pom.xml]: sure, can we just group it with the other non test scope dependencies then? for my OCD :)

Comment (allanmckenzie) [reply framework-utilities/test-utils/test-utils-framework-api/pom.xml]: Good point. Done

**Review by Morbu — APPROVED** [review]

**Pull request merged**